### ISSUE TITLE
Hangout history

### ISSUE THREAD

**Issue by morphis** [tiagosh/telepathy-hanging#1]
What about fetching the server-side stored hangout history and pushing that as existing messages into the chat history within telepathy? Does telepathy support something like this?

**Comment by tiagosh** [comment]
Yes. telepathy-hanging already do that. It recovers messages sent to you since the last login.
Telepathy does not support fetching old history on demand as you scroll the chat view. That will require changes in the telepathy interfaces.

**Comment by morphis** [comment]
Ah great.

One thing I really enjoy with the Android hangouts app is that it keeps the whole history since the beginning. If you say "telepathy interfaces" you mean the official telepathy defined interfaces, so a change that would need changes in clients too?

**Comment by tiagosh** [comment]
Both. Telepathy interfaces should have some method like "getMoreMessages(string startEvent, uint suggestedAmount)" , and clients should call it when the user scroll the view to the top.

**Comment by morphis** [comment]
Sounds like a larger amount of work.

**Comment by tiagosh** [comment]
We could do it through custom interfaces for now. That is easy. We just create the interface and plug it into the TextChannel, and clients that wish to use the feature can implement/call it by hand instead of using telepathy-qt/telepathy-glib. Once it is stable, we can propose it upstream. I did something similar in telepathy-ofono.

**Comment by morphis** [comment]
Sounds like a good idea.
